src/components/lava-coder/lava-coder-view.tsx

```diff
@@ -104,12 +119,12 @@ export const LavaCoderView = observer(function LavaCoderView({ width, height, ma
                 "elevation:", `${Math.round(elevation)}m = ${elevationFeet}ft`,
                 "in hazard zone:", isInHazardZone);
     if (isPlaceVentMode && isInHazardZone) {
-      setVentLocation(latitude, longitude, elevation / verticalExaggeration);
+      lavaSimulation.setVentLocation(latitude, longitude, elevation / verticalExaggeration);
```

Comment:
With this change, the simulation automatically updates its vent location on vent placement. But the blockly block won't update its vent location value automatically, so unless the user manually pastes the new location in the block, that value will override the pinned location when the simulation is run. I guess that's how things work right now anyway, but I really hope we can figure out a better solution for all this at some point.

Reply:
In other words, previously there were _three_ independent representations of the vent location: the model, React state, and the Blockly block. This PR reduces it to two by eliminating the React state, but you're right there's still the Blockly block to consider. But now that I think about it, I think that's actually the right behavior. The Blockly blocks represent the "program" the student wrote. That program sets the model state when it is run. Since the story specifies that reload doesn't affect the Blockly blocks, I think this is the right behavior, even if it's potentially confusing.